tests/dragonfly/replication_test.py

```diff
@@ -578,30 +580,47 @@ async def test_expiry(df_local_factory, n_keys=1000):
     res = await c_replica.mget(k for k, _ in gen_test_data(n_keys))
     assert all(v is not None for v in res)
 
-    # Set key expries in 500ms
+    # Set key differnt expries times in ms
     pipe = c_master.pipeline(transaction=True)
     for k, _ in gen_test_data(n_keys):
-        pipe.pexpire(k, 500)
+        ms = random.randint(100, 500)
+        pipe.pexpire(k, ms)
     await pipe.execute()
 
-    # Wait two seconds for heatbeat to pick them up
-    await asyncio.sleep(2.0)
-
-    assert len(await c_master.keys()) == 0
-    assert len(await c_replica.keys()) == 0
+    # send more traffic for differnt dbs while keys are expired
+    for i in range(8):
+        is_multi = i % 2
+        c_master_db = aioredis.Redis(port=master.port, db=i)
+        pipe = c_master_db.pipeline(transaction=is_multi)
+        # Set simple keys n_keys..n_keys*2 on master
+        start_key = n_keys*(i+1)
+        end_key = start_key + n_keys
+        batch_fill_data(client=pipe, gen=gen_test_data(
+            end_key, start_key), batch_size=20)
+
+        await pipe.execute()
```

Comment:
you could run this in parallel with `asyncio crate_task`, I assume this would be more reliable

Reply:
but actually I do not want this to be in parallel to make sure I add entries while other entries are expired, if I will do it in parallel it might be that I finished adding the entries before heartbeat expired the above entries